src/core/search/rax_tree_test.cc

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/rax_tree.h"
+
+#include <absl/container/btree_set.h>
+#include <absl/strings/str_cat.h>
+#include <gtest/gtest.h>
+#include <mimalloc.h>
+
+#include <algorithm>
+#include <memory_resource>
+
+#include "base/gtest.h"
+#include "base/logging.h"
+
+extern "C" {
+#include "redis/zmalloc.h"
+}
+
+namespace dfly::search {
+
+using namespace std;
+
+struct RaxTreeTest : public ::testing::Test {
+  static void SetUpTestSuite() {
+    auto* tlh = mi_heap_get_backing();
+    init_zmalloc_threadlocal(tlh);
+  }
+};
+
+TEST_F(RaxTreeTest, EmplaceAndIterate) {
+  RaxTreeMap<std::string> map(pmr::get_default_resource());
+
+  vector<pair<string, string>> elements(90);
+  for (int i = 10; i < 100; i++)
+    elements[i - 10] = make_pair(absl::StrCat("key-", i), absl::StrCat("value-", i));
+
+  for (auto& [key, value] : elements) {
+    auto [it, inserted] = map.try_emplace(key, value);
+    EXPECT_TRUE(inserted);
+    EXPECT_EQ(it->first, key);
+    EXPECT_EQ(it->second, value);
+  }
+
+  size_t i = 0;
+  for (auto [key, value] : map) {
+    EXPECT_EQ(elements[i].first, key);
+    EXPECT_EQ(elements[i].second, value);
+    i++;
+  }
+}
+
+TEST_F(RaxTreeTest, LowerBound) {
```

Comment:
check how it works with empty string_views, strings. Sometimes redis code has problems with zero length slices that have nullptr pointers